lib/seattleflu/id3c/cli/command/etl/redcap_det_fh_airs.py

```diff
@@ -0,0 +1,910 @@
+    # they are populated they use the browser's time zone.
+    collection_date = None
+
+    if collection_method == CollectionMethod.KIOSK:
+        collection_date = extract_date_from_survey_timestamp(record, "kiosk_registration_4c7f") or record.get("nasal_swab_q")
+
+    elif collection_method in [CollectionMethod.SWAB_AND_SEND, CollectionMethod.UW_DROPBOX]:
+        collection_date = record.get("date_on_tube") \
+            or extract_date_from_survey_timestamp(record, "husky_test_kit_registration") \
+            or record.get("kit_reg_date") \
+            or extract_date_from_survey_timestamp(record, "test_fulfillment_form") \
+            or record.get("back_end_scan_date")
+
+    return collection_date
+
+
+def airs_create_enrollment_questionnaire_response(record: REDCapRecord, patient_reference: dict,
+    encounter_reference: dict) -> Optional[dict]:
+    """
+    Returns a FHIR Questionnaire Response resource entry for the enrollment
+    encounter (i.e. encounter of enrollment into the study)
+    """
+
+    integer_questions = [
+        'scr_age',
+        'scr_num_doses'
+        'scr_vacc_infl',
+        'enr_outside_hour',
+        'enr_outside_mask',
+        'enr_sex',
+        'enr_gender',
+        'enr_degree',
+        'enr_ethnicity',
+        'enr_living_group',
+        'enr_living_population',
+        'enr_outside_mask',
+        'enr_indoor_no_mask',
+        'enr_outdoor_no_mask',
+    ]
+
+    string_questions = [
+        'scr_dose1_man',
+        'scr_dose1_man_other',
+        'scr_dose2_man',
+        'scr_dose2_man_other',
+        'scr_dose3_man',
+        'scr_dose3_man_other',
+        'enr_other_race',
+        'enr_other_group',
+    ]
+
+    date_questions = [
+        'scr_dose1_date',
+        'scr_dose2_date',
+        'scr_dose3_date',
+        'scr_vacc_infl_date',
+    ]
+
+    boolean_questions = [
+        'scr_contact_pub',
+        'enr_living_alone',
+        'enr_living_area',
+    ]
+
+    decimal_questions: List[str] = [
+        # TODO: none so far, but maybe in future
+    ]
+
+    coding_questions = [
+        'enr_race'
+        'enr_mask_type'
+    ]
+
+    # Do some pre-processing
+    # Combine checkbox answers into one list
+    checkbox_fields = [
+        'enr_race',
+        'enr_mask_type'
+    ]
+
+    question_categories = {
+        'valueCoding': coding_questions,
+        'valueBoolean': boolean_questions,
+        'valueInteger': integer_questions,
+        'valueString': string_questions,
+        'valueDate': date_questions,
+        #'valueDecimal': decimal_questions (none so far)
+    }
+
+    for field in checkbox_fields:
+        record[field] = combine_checkbox_answers(record, field)
+
+    # Combine all fields answering the same question
+    record['countries_visited_base'] = combine_multiple_fields(record, 'country', '_base')
+    record['states_visited_base'] = combine_multiple_fields(record, 'state', '_base')
```

Comment:
Are these questions asked in AIRS? They caught my eye as perhaps questions specific to HCT.

Reply:
Noted and removed.